activiti-core/activiti-bpmn-converter/src/main/java/org/activiti/bpmn/converter/child/LinkEventSourceParser.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2010-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.bpmn.converter.child;
+
+import org.activiti.bpmn.model.BaseElement;
+import org.activiti.bpmn.model.BpmnModel;
+import org.activiti.bpmn.model.LinkEventDefinition;
+
+import javax.xml.stream.XMLStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LinkEventSourceParser extends BaseChildElementParser{
+
+    public String getElementName() {
+        return ATTRIBUTE_LINK_SOURCE;
+    }
+
+    public void parseChildElement(XMLStreamReader xtr, BaseElement parentElement, BpmnModel model) throws Exception {
+        if (!(parentElement instanceof LinkEventDefinition)) {
+            return;
+        }
+
+        LinkEventDefinition eventDefinition = (LinkEventDefinition) parentElement;
+        List<String> sources=new ArrayList<>();
+        sources.add(xtr.getElementText());
+        eventDefinition.setSource(sources);
+    }
```

Comment:
This can be simplified using [pattern matching](https://docs.oracle.com/en/java/javase/18/language/pattern-matching-instanceof-operator.html) to avoid casting on `instanceof`.

```suggestion
        if (parentElement instanceof LinkEventDefinition linkEventDefinition) {
            List<String> sources = new ArrayList<>();
            sources.add(xtr.getElementText());
            linkEventDefinition.setSource(sources);
        }
```

Reply:
Done